internal/runners/eval/eval.go

```diff
@@ -0,0 +1,92 @@
+package eval
+
+import (
+	"strings"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runbits/rationalize"
+	"github.com/ActiveState/cli/pkg/localcommit"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildscript"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type primeable interface {
+	primer.Outputer
+	primer.Auther
+	primer.Projecter
+}
+
+type Params struct {
+	Target string
+}
+
+type Eval struct {
+	out     output.Outputer
+	project *project.Project
+	auth    *authentication.Auth
+}
+
+type errTargetNotFound struct {
+	error
+	target string
+}
+
+func New(p primeable) *Eval {
+	return &Eval{
+		out:     p.Output(),
+		project: p.Project(),
+		auth:    p.Auth(),
+	}
+}
+
+func (e *Eval) Run(params *Params) (rerr error) {
+	defer rationalizeError(e.auth, e.project, &rerr)
+
+	if !e.auth.Authenticated() {
+		return rationalize.ErrNotAuthenticated
+	}
+
+	if e.project == nil {
+		return rationalize.ErrNoProject
+	}
+
+	commitID, err := localcommit.Get(e.project.Dir())
+	if err != nil {
+		return errs.Wrap(err, "Unable to get commit ID")
+	}
+
+	script, err := buildscript.ScriptFromProject(e.project)
+	if err != nil {
+		return errs.Wrap(err, "Could not get local build script")
+	}
+
+	var target string
+	for _, assignment := range script.Assignments {
+		if strings.EqualFold(assignment.Key, params.Target) {
+			target = assignment.Key
+		}
+	}
+
+	if target == "" {
+		return errTargetNotFound{target: params.Target}
+	}
```

Comment:
This should be the job of the API to validate. This feels overly opinionated to validate this client side, we should just ensure that we can communicate the API error that this would result in.

Reply:
The API error in question has messaging around build expressions, not build scripts. Do we want to forward this to the user? The error in question is a planning error which can have different message contents so I believe it would come down to string matching to determine the error type.

Here's the error I got: `Unable to find requested target 'thing' in build expression`

Reply:
I'm fine with the string matching approach if that's what we want to do for now. We can file a follow-up to get a more specific error for this case from the API.

Reply:
Just communicating the API error alone is enough for now. But yeah we should file something with them so we can programmatically assert this error and give it our own message.

Reply:
Follow up filed here: https://activestatef.atlassian.net/browse/PB-4929